app/javascript/components/action-dialog/component.spec.js

```diff
@@ -0,0 +1,97 @@
+import { mountedComponent, screen } from "test-utils";
+import ActionDialog from "./component";
+
+describe("<ActionDialog />", () => {
+  const props = {
+    cancelHandler: () => { },
+    children: [],
+    confirmButtonLabel: "",
+    confirmButtonProps: {},
+    dialogActions: 'Test',
+    dialogSubHeader: "Test SubHeader",
+    dialogSubtitle: "Test Subtitle",
+    dialogText: "",
+    dialogTitle: "Test Title",
+    disableActions: false,
+    disableBackdropClick: false,
+    maxSize: "sm",
+    omitCloseAfterSuccess: false,
+    onClose: () => { },
+    open: true,
+    pending: false,
+    successHandler: () => { }
+  };
+
+  beforeEach(() => {
+    mountedComponent(<ActionDialog {...props} />);
+  });
+
+  it("should render Dialog", () => {
+    expect(screen.getAllByRole('dialog')).toHaveLength(1);
+  });
+
+  it("should render DialogTitle", () => {
+    expect(screen.getByRole("dialog")).toHaveTextContent("Test Title");
+  });
+
+  it("should render Submit Button", () => {
+    const newProps = {
+      ...props,
+      className: "MuiSvgIcon-root"
+    };
+    mountedComponent(<ActionDialog {...newProps} />);
+    expect(screen.getAllByRole('button', { className: 'MuiSvgIcon-root' })).toBeTruthy();
```

Comment:
Is there mutiple buttons here? If so figure out the text of the button using screen.debug() and query with getByText instead. Also use toBeInTheDocument

Reply:
yes there are multiple submit button and they does not have submit text instead of that they are having icons and all submit button have same icons and classes